chainerx_cc/chainerx/python/routines.cc

```diff
@@ -315,26 +315,39 @@ void InitChainerxEvaluation(pybind11::module& m) {
 void InitChainerxIndexing(pybind11::module& m) {
     // indexing routines
     m.def("take",
-          [](const ArrayBodyPtr& a, py::handle indices, const absl::optional<int8_t>& axis) {
-              if (!axis.has_value()) {
-                  throw NotImplementedError{"axis=None is not yet supported for chainerx.take."};
```

Comment:
We must keep this check (it should be tested but that's another story)?

Reply:
It is now checked in the cuda implementation

https://github.com/chainer/chainer/pull/8197/files#diff-effbca5a87c4ee8ecee4fc28fd587b8dR225

I wanted to abstract specific implementation details from the binding code

Reply:
Yes, I dont know why I deleted it